v2/dtos/requests/deviceservice.go

```diff
@@ -60,3 +62,121 @@ func AddDeviceServiceReqToDeviceServiceModels(addRequests []AddDeviceServiceRequ
 	}
 	return DeviceServices
 }
+
+// UpdateDeviceServiceRequest defines the Request Content for PUT event as pushed DTO.
+// This object and its properties correspond to the UpdateDeviceServiceRequest object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/UpdateDeviceServiceRequest
+type UpdateDeviceServiceRequest struct {
+	common.BaseRequest `json:",inline"`
+	Service            dtos.UpdateDeviceServiceDTO `json:"service"`
+}
+
+// Validate satisfies the Validator interface
+func (ds UpdateDeviceServiceRequest) Validate() error {
+	err := v2.Validate(ds)
+	return err
+}
+
+// UnmarshalJSON implements the Unmarshaler interface for the UpdateDeviceServiceRequest type
+func (ds *UpdateDeviceServiceRequest) UnmarshalJSON(b []byte) error {
+	var alias struct {
+		common.BaseRequest
+		Service dtos.UpdateDeviceServiceDTO
+	}
+	if err := json.Unmarshal(b, &alias); err != nil {
+		return v2.NewErrContractInvalid("Failed to unmarshal request body as JSON.")
+	}
+
+	*ds = UpdateDeviceServiceRequest(alias)
+
+	// validate AddDeviceServiceRequest DTO
+	if err := ds.Validate(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ReplaceDeviceServiceModelFieldsWithDTO replace model's fields with DTO patch
+func ReplaceDeviceServiceModelFieldsWithDTO(ds *models.DeviceService, patch dtos.UpdateDeviceServiceDTO) {
```

Comment:
So idea is the DB update will load existing `DeviceService` model from DB, call this to update the fields that changed and then store the update  `DeviceService` model to DB?

Reply:
Yes, I thought the metadata handler will invoke dbClient to load the existing DeviceService and replace the field value with patch, then store into the DB.